### PR TITLE
feat: Enhance `RemoteServer` and `Console` create-user to assign group

### PR DESCRIPTION
**Pull request by gramian**
## What does this PR do?

1. The `RemoteServe` class method `createUser` is overloaded with a variant with the signature
```
public void createUser(final String userName, final String password, final HashMap<String,String> databases)
```
As for the HTTP API server command the `databases` HashMap should have keys being database names and values being a group name. 
The former function with a `List<String>` `databases` argument is still available, but its body changed so that the new method is called by it.

2. The `Console` class method `executeCreateUser` is updated so the new `RemoteServer` `createUser` method is used. However not too robustly, and error handling is defered to the `RemoteServer` or eventually to the `ServerSecurityUser` class.

So now in console the following can be done:
```
CREATE USER user IDENTIFIED BY password GRANT CONNECT TO db:group
```

## Motivation

Missing capability to assign group during user creation in console, even though possibl via HTTP API.

## Additional Notes

It is possible to pass an empty group, but AFAIK this is also allowed in the HTTP server command.

## Checklist

- [X] I have run the build using `mvn clean package` command
- [ ] My unit tests cover both failure and success scenarios

### CODE REVIEW

File: console/src/main/java/com/arcadedb/console/Console.java [hunk 1]
```diff
@@ -467,16 +467,25 @@ private void executeCreateUser(final String params) {
     checkHasSpaces("User name", userName);
 
     final String password;
-    final List<String> databases;
+    HashMap<String,String> databases = new HashMap<String,String>();
```
Comment (robfrank) [inline console/src/main/java/com/arcadedb/console/Console.java]: my suggestion:
``` final Map<String,String>...```

File: network/src/main/java/com/arcadedb/remote/RemoteServer.java [hunk 2]
```diff
@@ -74,7 +74,7 @@ public String toString() {
     return protocol + "://" + currentServer + ":" + currentPort;
   }
 
-  public void createUser(final String userName, final String password, final List<String> databases) {
+  public void createUser(final String userName, final String password, final HashMap<String,String> databases) {
```
Comment (robfrank) [inline network/src/main/java/com/arcadedb/remote/RemoteServer.java]: again, prefer interface:
```final Map<String,String```

File: network/src/main/java/com/arcadedb/remote/RemoteServer.java [hunk 3]
```diff
@@ -101,6 +101,16 @@ public void createUser(final String userName, final String password, final List<
     }
   }
 
+  public void createUser(final String userName, final String password, final List<String> databases) {
+
+    HashMap<String,String> databasesWithGroups = new HashMap<String, String>();
```
Comment (robfrank) [inline network/src/main/java/com/arcadedb/remote/RemoteServer.java]: again:
```Map<String,String> databasesWithGroups = new HashMap<String, String>();```

**Comment by lvca** [comment]
Good for me ;-)

**Review by robfrank — APPROVED** [review]
LGTM

**Pull request merged**